uji.py

```diff
@@ -1054,6 +1065,152 @@ def upload(self):
+            if in_section:
+                if '```' in line:
+                    in_section = False
+                else:
+                    # remove existing output
+                    continue
+            if section_index < 0:
+                # find the existing ``` markers
+                if '```' in line:
+                    in_section = True
+                    section_index = len(self.lines) + 1
+                # new section, enforce our markers, right below cursor_offset
+                if self.is_checkbox(line):
+                    section_index = self._insert(self.cursor_offset + 1, '```\n')
+                    self._insert(section_index, '```\n')
+            # rest is going to be forwarded as it was
+            self.lines.append(line)
+
+        # return the index where to store the output of the command
+        return section_index
+
+    def _sanitize(self, line):
+        return line.replace('\t', '    ')
+
+    def execute_command(self):
+        line = self.lines[self.cursor_offset]
+        if not self.is_checkbox(line) or "⚙" not in line:
+            return
+
+        # list of regex to deduce the type of command to be run
+        command_re = {
+                'attach': r'^(\s*)- \[.\].* \[`(.*)`\]\((.*)\).*',
+                'human': r'^(\s*)- \[.\].* `(.*)`: <strong>(.*)</strong>$',
+                'single': r'^(\s*)- \[.\].* `(.*)`: `(.*)`$',
+                'multi': r'^(\s*)- \[.\].* `(.*)`:$',
+                'exitcode': r'^(\s*)- \[.\].* `(.*)`$',
+        }
+
+        command_type = None
+        command_match = None
+
+        for type_c, re_c in command_re.items():
+            match = re.match(re_c, line)
+            if match:
+                command_type = type_c
+                command_match = match
+                break
+
+        if command_type is None:
+            logger.error(f'Failed to match run command line: {line}')
+            return
+
+        command = match[2]
+        output = []
+        insert_offset = self.cursor_offset + 1
+        section_index = -1
+
+        if command_type == 'multi':
+            section_index = self.clean_multi_lines_code()
+
+        try:
+            import subprocess
+
+            proc = subprocess.Popen(command, shell=True, universal_newlines=True,
+                                    bufsize=0,
+                                    stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+
+            while proc.poll() is None:
+                if command_type == 'multi':
+                    while line := proc.stdout.readline():
+                        section_index = self._insert(section_index, line)
+                        output.append(line)
+                        self.writeout()
+                        self._redraw()
+                        self._render()
+                else:
+                    output.extend(proc.stdout.readlines())
+            result = proc.poll()
+        except KeyboardInterrupt:
+            proc.kill()
```

Comment:
```suggestion
            proc.terminate()
            time.sleep(0.2)
            if proc.poll() is None:
                   proc.kill()
```
for a nicer shutdown?